components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java

```diff
@@ -85,6 +88,19 @@ public void startRoute(String id) throws Exception {
         camelContext.getRouteController().startRoute(id);
     }
 
+    public String resetRoute(String id) throws Exception {
+        ManagedRouteMBean managedRouteMBean = camelContext.getManagedRoute(id, ManagedRouteMBean.class);
+        if (managedRouteMBean != null) {
+            managedRouteMBean.reset(true);
+        } else {
+            String description = "The provided route id does not exists :" + camelContext.getRoute(id);
+            ResponseEntity<?> responseEntity = ResponseEntity.status(HttpStatus.BAD_REQUEST).body(description);
+            ObjectMapper objectMapper = new ObjectMapper();
+            return objectMapper.writeValueAsString(responseEntity);
+        }
+        return null;
```

Comment:
Not sure if this method should return anything as in fact in the Mvc endpoint is used for logging purpose only